source/CommonJobs/CommonJobs.Domain/Applicant.cs

```diff
@@ -41,9 +41,9 @@ public void AddGeneralNote(string note, AttachmentReference attachment = null)
             });
         }
 
-        public override IEnumerable<SlotWithAttachment> AllAttachmentReferences
+        public override SlotWithAttachment[] AllAttachmentReferences
         {
-            get { return base.AllAttachmentReferences.Union(SlotWithAttachment.GenerateFromNotes(Notes)); }
+            get { return base.AllAttachmentReferences.Union(SlotWithAttachment.GenerateFromNotes(Notes)).ToArray(); }
```

Comment:
`ToArray` will force the materialization of an enumeration, but `IEnumerable<>` will not, until iterated. Why the change?